### PR TITLE
Error log in

### PR DESCRIPTION
**Pull request by mikeramz86**
@evilsoft 

# Summary
1. Created new branch called errorLogIn
1. Changed necessary px to rems
1. Added evilsoft's css defaults
1. Created an error object using propTypes and defaultprops

# Notes
1. Not currently working
1. I know I am missing a piece of code to see it in action and tried figuring out what it was. I don't remember
1. Reviewing propTypes and defaultProps this week

### CODE REVIEW

File: notes.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+console.log()
+
+
+# Notes
+1. make sure your files are named after the components
+...1. 
+1. always push with padding and not margin PADDING ALL THE TIME!!!!
+1. Duck typing
+1. If
```
Comment (evilsoft) [inline notes.md]: Did not finish your thoughts before committing

File: src/LoginPage.css [hunk 2]
```diff
@@ -1,43 +1,33 @@
...
-    margin: 0;
-    font-size: 18px;
-  } */
-  
-  body * {
-    box-sizing: border-box;
-  }
-  
-  fieldset {
+/* fieldset {
```
Comment (evilsoft) [inline src/LoginPage.css]: Should not commit commented code.

File: src/LoginPage.css [hunk 3]
```diff
@@ -46,19 +36,19 @@
...
-    font-size: 18px;
+    font-size: 1rem;
     text-align: left;
-    padding: 15px;
+    padding: 1rem;
     border: 2px solid #e7e7e7;
     border-radius: 3px;
     background: #fcfcfc;
-    margin: 10px auto 20px;
+    margin: 10px auto 2rem;
```
Comment (evilsoft) [inline src/LoginPage.css]: why not `rem` for the first: `1rem` === `10px`

File: src/LoginPage.js [hunk 4]
```diff
@@ -46,21 +50,26 @@ class LoginPage extends Component {
...
+        { message && type === 'loginError' && <Flash message={message}/> }
         <form onSubmit={this.handleSubmit}>
           <fieldset>
             {this.state.error && (
               <h3 data-test="error" onClick={this.dismissError}>
                 <button onClick={this.dismissError}>✖</button>
                 {this.state.error}
               </h3>
-            )}
+            )}
```
Comment (evilsoft) [inline src/LoginPage.js]: trailing whitespace
👾

File: src/LoginPage.js [hunk 5]
```diff
@@ -46,21 +50,26 @@ class LoginPage extends Component {
...
                 <button onClick={this.dismissError}>✖</button>
                 {this.state.error}
               </h3>
-            )}
+            )}            
             <label>User Name</label>
             <input
               type="text"
-              data-test="username"
+              data-test="userame"
```
Comment (evilsoft) [inline src/LoginPage.js]: think there may a spelling err here.
Also curious about these data attributes

File: src/index.css [hunk 6]
```diff
@@ -1,14 +1,24 @@
...
+  font-family:  sans-serif;
+  font-size:    unit(1.6em);
+  font-weight:  400;
+  line-height:  1.25;
+}
+
+article, aside, details, figcaption, figure,
+footer, header, hgroup, menu, nav, section {
+  display: block;
+}
```
Comment (evilsoft) [inline src/index.css]: Missing a newline here

File: src/Greeting.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+Greeting.propTypes = {
+  name: PropTypes.string
+};
+
+Greeting.defaultProps = {
+  name: ""
+};
+
+export default Greeting;
```
Comment (evilsoft) [inline src/Greeting.js]: Missing a newline

File: src/Greeting.css [hunk 8]
```diff
@@ -0,0 +1,9 @@
+h3 {
+    font-size: 2em;
+    margin-left: 1.25rem;
+    margin-right: 1.25rem;
+    padding: 1rem;
+    border-left: 10px solid #0E59B2;
+    border-radius: 2.5px;
+    background-color: lightblue;
+}
```
Comment (evilsoft) [inline src/Greeting.css]: Missing a newline here

File: src/Greeting.css [hunk 9]
```diff
@@ -0,0 +1,9 @@
+h3 {
+    font-size: 2em;
+    margin-left: 1.25rem;
+    margin-right: 1.25rem;
+    padding: 1rem;
+    border-left: 10px solid #0E59B2;
+    border-radius: 2.5px;
```
Comment (evilsoft) [inline src/Greeting.css]: want to avoid partial pixels for some displays and browsers

File: src/Greeting.js [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+  }
+}
+
+Greeting.propTypes = {
+  name: PropTypes.string
+};
+
+Greeting.defaultProps = {
+  name: ""
+};
```
Comment (evilsoft) [inline src/Greeting.js]: unintended semicolon?

File: src/Greeting.js [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+
+class Greeting extends Component {
+  render() {
+    return <h3>Hello, {this.props.name}</h3>
+  }
+}
+
+Greeting.propTypes = {
+  name: PropTypes.string
+};
```
Comment (evilsoft) [inline src/Greeting.js]: unintended semicolon?

**Comment by evilsoft** [comment]
As part of this, PR update the font-sizes to be the default browser sizes. for labels, texts and buttons.

**Comment by mikeramz86** [comment]
Just for clarification, when you say to update the font-sizes to the default browser sizes does that mean change all font-sizes in all the style sheets to 1rem? For example for those font sizes that are 2rem set them to 1rem?

**Comment by evilsoft** [comment]
Sorry about that the default browser font-size (which is 16px).

**Comment by mikeramz86** [comment]
Sorry, do you mean me to do this for everything?
```
font-size: 16px;
```

instead of:

```
font-size: 1rem;
```

**Comment by evilsoft** [comment]
Ah. `1rem !== 16px`.
`1.6rem === 16px`
so, `font-size: 1.6rem` matches the browser defaults of `16px

**Pull request merged**